challenge_spr/src/challenge_spr_states/riddle_game.py

```diff
@@ -25,141 +26,162 @@ def hear(robot, time_out):
 def answer(robot, res, crowd_data):
     if res:
         if 'actions' in res.semantics:
+            answer = "I don't know"
+
             for action in res.semantics['actions']:
+                try:
+                    if 'action' not in action:
+                        continue
+
+                    answer = "Sorry, I cannot %s. I don't know the answer.  " % str(action['action'])
+
+                    if action['action'] == 'random_gender':
+                        answer = random.choice(["A male", "A female"])
+
+                    # Predefined questions
+                    if action['action'] == 'answer':
+                        # if res.semantics['actions'] == 'answer':
+                        answer = action['solution']#res.semantics['solution']
+
+                    # Counting people
+                    if action['action'] == 'count':
+
+                        if action['entity'] == 'people':
+                            answer = 'In the crowd are %d people' % crowd_data['crowd_size']
 
-                # Predefined questions
-                if action['action'] == 'answer':
-                    # if res.semantics['actions'] == 'answer':
-                    answer = action['solution']#res.semantics['solution']
+                        if action['entity'] == 'children':
+                            answer = 'In the crowd are %d children' % crowd_data['children']
 
-                # Counting people
-                if action['action'] == 'count':
+                        if action['entity'] == 'adults':
```

Comment:
@SamAlexandrov We might cook up a more elegant solution here. Not necessarily in this merge request though